src/components/BottomSheetModal/index.tsx

```diff
@@ -0,0 +1,117 @@
+import * as React from "react";
+import { BackHandler, StyleSheet } from "react-native";
+import {
+  BottomSheetModal as RNBottomSheetModal,
+  BottomSheetBackdrop,
+  BottomSheetView,
+  useBottomSheetDynamicSnapPoints,
+} from "@gorhom/bottom-sheet";
+
+const INITIAL_SNAP_POINTS = ["CONTENT_HEIGHT"];
+
+export function useBottomSheetModal({ openOnMount }: { openOnMount: boolean }) {
+  const initiallyOpenedRef = React.useRef(false);
+  const sheetRef = React.useRef<RNBottomSheetModal>(null);
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  const closeSheet = React.useCallback(() => {
+    if (sheetRef.current) {
+      sheetRef.current.dismiss();
+      setIsOpen(false);
+    }
+  }, []);
+
+  const openSheet = React.useCallback(() => {
+    if (sheetRef.current) {
+      sheetRef.current.present();
+      setIsOpen(true);
+    }
+  }, []);
+
+  React.useEffect(() => {
+    if (!initiallyOpenedRef.current && openOnMount) {
+      initiallyOpenedRef.current = true;
+      openSheet();
+    }
+  }, [openOnMount, openSheet]);
+
+  return { sheetRef, closeSheet, openSheet, isOpen };
+}
+
+function useBackPressHandler(onHardwareBackPress?: () => void | boolean) {
+  React.useEffect(() => {
+    const onBack = () => {
+      if (onHardwareBackPress) {
+        const backPress = onHardwareBackPress();
+        if (typeof backPress === "boolean") {
+          return backPress;
+        }
+      }
+
+      // We don't allow the back press to navigate/dismiss this modal by default
+      return true;
+    };
+
+    BackHandler.addEventListener("hardwareBackPress", onBack);
+
+    return () => BackHandler.removeEventListener("hardwareBackPress", onBack);
+  }, [onHardwareBackPress]);
+}
```

Comment:
I think we should block the back interaction using this: https://reactnavigation.org/docs/preventing-going-back

So instead of `onHardwareBAckPress` we use `onBackAction` just so it catches swiping, the back button, and the ios back button

Reply:
ah yeah good point. addressed via https://github.com/digidem/mapeo-mobile-shell/pull/27/commits/75916955a10bd21df66f9996f1a72fb968021ab2